src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -430,6 +478,62 @@
                              jruby-service)
         (environment-not-found environment)))))
 
+(defn exception-matches?
+  [^Exception e pattern]
+  (->> e
+       .getMessage
+       (re-find pattern)
+       boolean))
+
+(schema/defn ^:always-validate
+  task-details :- TaskDetails
+  [jruby-service :- (schema/protocol jruby-protocol/JRubyPuppetService)
+   jruby-instance
+   environment-name :- schema/Str
+   module-name :- schema/Str
+   task-name :- schema/Str]
+  (-> (jruby-protocol/get-task-data jruby-service
+                                    jruby-instance
+                                    environment-name
+                                    module-name
+                                    task-name)
+      sort-nested-info-maps
+      (describe-task environment-name module-name)))
+
+(schema/defn ^:always-validate
+  task-details-fn :- IFn
+  "Middleware function for constructing a Ring response from an incoming
+  request for detailed task information."
+  [jruby-service :- (schema/protocol jruby-protocol/JRubyPuppetService)]
+  (fn [request]
```

Comment:
Not sure if destructuring would help at all here. `{:keys [route-params jruby-instance] :as request}` or something. Might not be worth it.

Reply:
yeah, I'm not sure it's worth it